Common/Data/Encoding/Utf8.cpp

```diff
@@ -206,35 +206,22 @@ int u8_charnum(const char *s, int offset)
   return charnum;
 }
 
-/* number of characters */
-int u8_strlen(const char *s)
-{
-  int count = 0;
-  int i = 0;
-
-  while (u8_nextchar(s, &i) != 0)
-    count++;
-
-  return count;
-}
-
 /* reads the next utf-8 sequence out of a string, updating an index */
-uint32_t u8_nextchar(const char *s, int *index) {
+uint32_t u8_nextchar(const char *s, int *index, size_t size) {
 	uint32_t ch = 0;
 	int sz = 0;
 	int i = *index;
 	do {
 		ch = (ch << 6) + (unsigned char)s[i++];
 		sz++;
-	} while (s[i] && ((s[i]) & 0xC0) == 0x80);
+	} while (i < size && ((s[i]) & 0xC0) == 0x80);
```

Comment:
A string may have nulls in it which should terminate, even if they are earlier than the expected size?  I think we've even seen this with savedata.

-[Unknown]

Reply:
Hm, yes, you're right, this is a concern, I will add it back.

If a `string_view` is created from a `const char *`, it will already be truncated at the first null. But if you create it from an std::string that contains nulls, or explicitly with pointer range, it indeed can also contain nulls.